src/Microsoft.SymbolManifestGenerator/SymbolManifestGenerator.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.SymbolStore;
+using Microsoft.SymbolStore.KeyGenerators;
+
+namespace Microsoft.SymbolManifestGenerator
+{
+    public class SymbolManifestGenerator
+    {
+        private const int Private = 340;
+        public SymbolManifestGenerator()
+        {
+        }
+
+        public static void GenerateManifest(ITracer tracer, DirectoryInfo dir, string manifestFileName)
+        {
+            try
+            {
+                ManifestDataV1 manifestData = new();
+
+                // Iterate through all files recursively in the directory
+                foreach (FileInfo file in dir.GetFiles("*", SearchOption.AllDirectories))
+                {
+                    string relativePath = file.DirectoryName.Replace(dir.FullName, string.Empty).TrimStart(Path.DirectorySeparatorChar);
+                    FileStream fileStream = file.OpenRead();
+                    SymbolStoreFile symbolStoreFile = new(fileStream, file.FullName);
+                    FileKeyGenerator generator = new(tracer, symbolStoreFile);
+                    IEnumerable<SymbolStoreKey> clrKeys = generator.GetKeys(KeyTypeFlags.ClrKeys);
+                    foreach (SymbolStoreKey clrKey in clrKeys)
+                    {
+                        string additionalDebugEntryFileName = clrKey.FullPathName;
+                        string basedirRelativePath = Path.Combine(relativePath, additionalDebugEntryFileName);
+                        string additionalDebugEntryFilePath = Path.Combine(dir.FullName, basedirRelativePath);
+                        FileInfo additionalDebugEntryFileInfo = new(additionalDebugEntryFilePath);
```

Comment:
This would make more sense if this line was before the CalculateSHA512 call on line 45.

Reply:
Ah, after checking the file exists, thanks for the catch!